### PR TITLE
Cbo 1188 agfs hardship api

### PR DESCRIPTION
**Pull request by kmahern**
#### What

Added api endpoints for advocate hardship claims.

#### Ticket

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=222&modal=detail&selectedIssue=CBO-1188

#### Why

To allow external users (eventually) to submit advocate hardship claims with their case management systems via the the api endpoints.

#### How

Added POST /api/external_users/claims/advocates/hardship/validate and POST /api/external_users/claims/advocates/hardship endpoints.

### CODE REVIEW

File: spec/api/v1/dropdown_data_spec.rb [hunk 1]
```diff
@@ -63,13 +65,15 @@
...
-        TRANSFER_CASE_CONCLUSIONS_ENDPOINT => API::Entities::SimpleKeyValueList.represent(Claim::TransferBrain::CASE_CONCLUSIONS.to_a).to_json
+        TRANSFER_CASE_CONCLUSIONS_ENDPOINT => API::Entities::SimpleKeyValueList.represent(Claim::TransferBrain::CASE_CONCLUSIONS.to_a).to_json,
+        CASE_STAGE_ENDPOINT => API::Entities::CaseStage.represent(CaseStage.all).to_json,
       }
     end
 
     before do
       seed_fee_schemes
       seed_case_types
+      seed_case_stage
```
Comment (jsugarman) [inline spec/api/v1/dropdown_data_spec.rb]: my error, but this should be `seed_case_stages` (plural). you could add a commit to fix my typo

**Review by jsugarman — CHANGES_REQUESTED** [review]
Can you go through the check llist in the ticket and make sure you have ticked them all off
https://dsdmoj.atlassian.net/browse/CBO-1188


you can exclude `seed_helper` from `simplecov` too, to improve coverage score. There should not be a reduction in coverage ideally.

File: app/interfaces/api/entities/case_stage.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module API
+  module Entities
+    class CaseStage < BaseEntity
+      expose :id
```
Comment (jsugarman) [inline app/interfaces/api/entities/case_stage.rb]: we should not really be exposing `id` - vendors would then use it and rely on it. we should be enabling search by `unique_code` (which we can guarantee will not change)

Comment (kmahern) [reply app/interfaces/api/entities/case_stage.rb]: 👍 Have removed id from the case stage endpoint (also api users will enter unique code and not id when creating/validating a hardship claim).

File: spec/api/v1/dropdown_data_spec.rb (hunk 1 above)
Comment (kmahern) [reply spec/api/v1/dropdown_data_spec.rb]: 👍 have amended to seed_case_stages

**Comment by kmahern** [comment]
@jsugarman I have been through the comments and the checklist on the ticket and I think I have made the changes requested.

**Comment by kmahern** [comment]
Have removed db/seed_helper.rb from the scope of code climate to resolve issue with code coverage.

File: spec/api/v1/external_users/claim_spec.rb [hunk 3]
```diff
@@ -16,6 +16,9 @@
     /api/external_users/claims/advocates/supplementary
     /api/external_users/claims/advocates/supplementary/validate
 
+    /api/external_users/claims/advocates/hardship
```
Comment (jsugarman) [inline spec/api/v1/external_users/claim_spec.rb:19]: 👍 yes this is good for AGFS 
see LGFS equivalent https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3341/files/d71bea25dc2edb4144f9e7fd772c502d1bb515fc#r420650769

**Pull request closed without merging**